Source/SpatialGDK/Private/EngineClasses/SpatialNetDriver.cpp

```diff
@@ -93,32 +92,63 @@ void USpatialNetDriver::OnMapLoaded(UWorld* LoadedWorld)
 
 	UE_LOG(LogSpatialOSNetDriver, Log, TEXT("Loaded Map %s. Connecting to SpatialOS."), *LoadedWorld->GetName());
 
-	checkf(!SpatialOSInstance->IsConnected(), TEXT("SpatialOS should not be connected already. This is probably because we attempted to travel to a different level, which current isn't supported."));
+	//checkf(!SpatialOSInstance->IsConnected(), TEXT("SpatialOS should not be connected already. This is probably because we attempted to travel to a different level, which current isn't supported."));
 
 	// Set the timer manager.
 	TimerManager = &LoadedWorld->GetTimerManager();
 
 	// Connect to SpatialOS.
-	SpatialOSInstance->ApplyConfiguration(WorkerConfig);
-	SpatialOSInstance->Connect();
+	Connect();
 
 	// Set up manager objects.
 	EntityRegistry = NewObject<UEntityRegistry>(this);
-	Interop = NewObject<USpatialInterop>(this);
+}
+
+void USpatialNetDriver::Connect()
+{
+	if (Connection != nullptr)
+	{
+		return;
+	}
+
+	Worker_ConnectionParameters Params = Worker_DefaultConnectionParameters();
+	Params.worker_type = TCHAR_TO_UTF8(*WorkerConfig.SpatialGDKApplication.WorkerPlatform);
+	Params.network.tcp.multiplex_level = 4;
+
+	Worker_ComponentVtable DefaultVtable = {};
+	Params.default_component_vtable = &DefaultVtable;
+
+	std::string Hostname = "127.0.0.1";
+	std::uint16_t Port = 7777;
+	std::string WorkerId = std::string(TCHAR_TO_UTF8(*WorkerConfig.SpatialGDKApplication.WorkerPlatform)) + std::string(TCHAR_TO_UTF8(*FGuid::NewGuid().ToString()));
+
+	// Start connecting
+	Worker_ConnectionFuture* ConnectionFuture = Worker_ConnectAsync(Hostname.c_str(), Port, WorkerId.c_str(), &Params);
+
+	// Wait for connection
+	Connection = Worker_ConnectionFuture_Get(ConnectionFuture, nullptr);
+	Worker_ConnectionFuture_Destroy(ConnectionFuture);
+
+	if (Worker_Connection_IsConnected(Connection))
+	{
+		OnSpatialOSConnected();
+	}
+	else
+	{
+		UE_LOG(LogSpatialOSNetDriver, Error, TEXT("Failed to connect to SpatialOS"));
```

Comment:
You should call `GetOpList` here and look for a `DisconnectOp`. That contains the failure reason.